Gradle test is not running on JDK 20

### Description of the bug

after the recent Gradle wrapper upgrade, build fails when running gradle test on JDK 20

```
> Task :compileFunctionalTestKotlin FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileFunctionalTestKotlin'.
> 'compileFunctionalTestJava' task (current target is 17) and 'compileFunctionalTestKotlin' task (current target is 20) jvm target compatibility should be set to the same Java version.
  Consider using JVM toolchain: https://kotl.in/gradle/jvm/toolchain
```

### Expected behavior

test passes on JDK 20

### Minimal reproducible example

checkout the current main branch (ccbe36855c6501e021478314f3dd567945a998c5)

```
- mvn install -DskipTests
- cd flow-plugins/flow-gradle-plugin
- ./gradlew build
```

### Versions

- Vaadin / Flow version: 24.2-SNASPHOT
- Java version: Java 20.0.2

--- a/flow-plugins/flow-gradle-plugin/build.gradle
+++ b/flow-plugins/flow-gradle-plugin/build.gradle
@@ -196,6 +196,9 @@ functionalTest {
 }
 
 kotlin {
+    jvmToolchain {
+        languageVersion.set(JavaLanguageVersion.of(17))
+    }
     explicitApi()
 }
 

